src/ng/parse.js

```diff
@@ -1698,12 +1698,14 @@ function $ParseProvider() {
   var cacheExpensive = createMap();
 
   this.$get = ['$filter', '$sniffer', function($filter, $sniffer) {
+    var csp = $sniffer.csp && !$sniffer.unsafeEval;
```

Comment:
Depending `csp` doesn't necessarily hit the `!$niffer.unsafeEval` check so this ensures that your `$sniffer.csp` is `true` and that `unsafeEval` is not allowed.